tests/conftest.py

```diff
@@ -0,0 +1,9 @@
+import pytest
+
+
+def pytest_collection_modifyitems(config, items):
```

Comment:
Any way we can use like `pytest_collection` or combo of `pytest_collect_file` to iterate through the collection to find  the parent module name and then iterate through each collection to assign the marker to each item?

That way rather than going through and renaming each test function to include the marker key it can be done so by module it belongs to? So any Items collect from `resources` are assigned its designated marker and the same for any items collected from `tasks`?

Reply:
Yeah, it's possible, I guess. One thing I've discussed with @btjd  is that I'm not a huge fan of this implementation because it's not explicit at all. Maybe - and I want to hear opinions here - we should choose the explicit path and mark each and every modules with it's own mark. For example:
tests/tasks/foo.py
```python
import pytest

pytestmark = pytest.mark.unit

def test_foo():
    pass
```
It's a lot more explicit - but verbose - than this "black magic" collection.

Reply:
I'm "lazy" so I'll take a little black magic sometimes :)

Valid point. I'm not apposed. Just another suggestion since you were already doing it I figured let's go one step further :)